kivy/core/image/__init__.py

```diff
@@ -644,10 +685,35 @@ def _set_anim_delay(self, x):
                 self._anim_ev = None
 
             if self._anim_delay >= 0:
-                self._anim_ev = Clock.schedule_interval(self._anim,
+                if not self.durations:
+                    self._anim_ev = Clock.schedule_interval(self._anim,
+                                                            self._anim_delay)
+                else:
+                    self._anim_ev = Clock.schedule_once(self._anim,
                                                         self._anim_delay)
 
     anim_delay = property(_get_anim_delay, _set_anim_delay)
+
+    def _get_durations(self):
+        return self._durations
+
+    def _set_durations(self, x):
+        # Waiting for list of ints or floats in microseconds like in GIF format, not seconds
+        if x is None:
+            self._durations = None
+            return
+        if not(isinstance(x, list) and all(isinstance(elem, (int, float)) for elem in x)) and len(x) > 0:
```

Comment:
better split the test over multiple lines for readability
```
        if not (
            isinstance(x, list)
            and all(isinstance(elem, (int, float)) for elem in x)
            and len(x) > 0
        ):
```
(also i think the closing parenthesis of the not mas misplaced).